api/src/main/java/io/lumeer/api/model/User.java

```diff
@@ -212,13 +208,42 @@ public String toString() {
             "id='" + id + '\'' +
             ", name='" + name + '\'' +
             ", email='" + email + '\'' +
-            ", authId='" + authIds + '\'' +
+            ", authIds=" + authIds +
             ", groups=" + groups +
             ", defaultWorkspace=" + defaultWorkspace +
             ", agreement=" + agreement +
             ", agreementDate=" + agreementDate +
             ", newsletter=" + newsletter +
+            ", wizardDismissed=" + wizardDismissed +
             ", wishes=" + wishes +
             '}';
    }
+
+   @Override
+   public boolean equals(final Object o) {
+      if (this == o) {
+         return true;
+      }
+      if (o == null || getClass() != o.getClass()) {
+         return false;
+      }
+      final User user = (User) o;
+      return Objects.equals(id, user.id) &&
+            Objects.equals(name, user.name) &&
+            Objects.equals(email, user.email) &&
+            Objects.equals(authIds, user.authIds) &&
+            Objects.equals(groups, user.groups) &&
+            Objects.equals(defaultWorkspace, user.defaultWorkspace) &&
+            Objects.equals(agreement, user.agreement) &&
+            Objects.equals(agreementDate, user.agreementDate) &&
+            Objects.equals(newsletter, user.newsletter) &&
+            Objects.equals(wizardDismissed, user.wizardDismissed) &&
+            Objects.equals(wishes, user.wishes);
+   }
+
+   @Override
+   public int hashCode() {
+      return Objects.hash(id, name, email, authIds, groups, defaultWorkspace, agreement, agreementDate, newsletter, wizardDismissed, wishes);
```

Comment:
Should not just `id` be used in `hashCode` and `equals` methods in such entities?

Reply:
Don't know, should it? What if it is not yet initialized and id is null?

Reply:
It should be some kind of a unique identifier. But you are right, `id` might be null. If `email` is not always provided either, then probably `authIds` is the best identifier. What do you think?